src/io/github/thebusybiscuit/cscorelib2/materials/MaterialCollections.java

```diff
@@ -53,17 +55,19 @@ private MaterialCollections() {}
 			
 			if (Tag.ITEMS_FISHES.isTagged(mat)) fishes.add(mat);
 			if (mat.toString().endsWith("_ORE")) ores.add(mat);
+			if (mat.toString().endsWith("_HEAD") || mat.toString().endsWith("_SKULL")) heads.add(mat);
 		}
 			
-		allLeaves = leaves.toArray(new Material[leaves.size()]);
-		allSaplings = leaves.toArray(new Material[saplings.size()]);
-		allLogs = leaves.toArray(new Material[logs.size()]);
-		allPlanks = leaves.toArray(new Material[planks.size()]);
-		allWools = leaves.toArray(new Material[wools.size()]);
-		allCarpets = leaves.toArray(new Material[carpets.size()]);
+		allLeaves = leaves.toArray(new Material[0]);
+		allSaplings = saplings.toArray(new Material[0]);
+		allLogs = logs.toArray(new Material[0]);
+		allPlanks = planks.toArray(new Material[0]);
+		allWools = wools.toArray(new Material[0]);
+		allCarpets = carpets.toArray(new Material[0]);
 		
-		allFishItems = fishes.toArray(new Material[fishes.size()]);
-		allOres = ores.toArray(new Material[ores.size()]);
+		allFishItems = fishes.toArray(new Material[0]);
+		allOres = ores.toArray(new Material[0]);
+		allHeads = heads.toArray(new Material[0]);
```

Comment:
You should provide the size of the list as the size of the array, not zero...

Reply:
Nope it's not necessary as toArray method creates a new array with the type of given array. If you make its size 0 its array is automatically sized to the size of the list.

Reply:
I know that it's not neccessary.
But you SHOULD still do so, using your method is just a bad practice because you are creating an array that isn't used anywhere and .toArray() has to create another array.
You are basically wasting resources by creating two arrays, one of which isn't even used there and just floats around in RAM for a small bit.